libraries/AP_AIS/AP_AIS.cpp

```diff
@@ -273,6 +273,25 @@ void AP_AIS::send(mavlink_channel_t chan)
     }
 }
 
+void AP_AIS::handle_message(const mavlink_ais_vessel_t &packet)
+{
+    if (!enabled()) {
+        return;
+    }
+
+    uint16_t index;
+    if (!get_vessel_index(packet.MMSI, index)) {
```

Comment:
You should pass the lat and lon here if there marked as valid. Otherwise the closer items will not take priority over further items.